### PR TITLE
Initial commit of metadata test.

### PR DESCRIPTION
**Pull request by kyle-long**
Test for metadata

### CODE REVIEW

File: spec/helpers/lib/manager.js [hunk 1]
```diff
@@ -28,56 +28,52 @@ module.exports = (config, fsAsync, logger, path, shelfLib, unique, URI) => {
...
+        return buildPath().then((testPath) => {
+            return new Space(reference, testPath);
         });
     }
 
 
     /**
-     * @param {string} file A path to a file.
-     * @return {Promise.<shelfLib~Artifact>}
+     * @see shelfFunctionalTests~Space.uploadArtifact
```
Comment (GoogilyBoogily) [inline spec/helpers/lib/manager.js:50]: Not a comment on actual code, but I didn't know about `@see`! That's cool.

File: spec/helpers/lib/space.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+"use strict";
+
+module.exports = (fsAsync, logger, path, unique) => {
+    /**
+     * This class exists to organize all artifacts within a
+     * certain test. For example, if I did a concurrency test
+     * where I uploaded many artifacts (or uploaded many metadatas)
```
Comment (GoogilyBoogily) [inline spec/helpers/lib/space.js]: "Metadatas" isn't a word.

File: spec/helpers/lib/space.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+     * they would all be contained within the same "testPath".
+     */
+    class Space {
+        /**
+         * testPath is the path to a directory that this Space
```
Comment (GoogilyBoogily) [inline spec/helpers/lib/space.js]: Seems like this description should be on the `@param` for `testPath` below.

File: spec/helpers/lib/space.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                return artifact.upload(content);
+            }).then((loc) => {
+                logger.debug(`Successfuly uploaded artifact to ${this.reference.buildUrl(loc)}`);
+                this.artifactList.push(artifact);
+
+                return artifact;
+            });
+        }
+
+        /**
```
Comment (GoogilyBoogily) [inline spec/helpers/lib/space.js]: Need another newline above this.

File: spec/metadata.spec.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+describe("METADATA", () => {
+    var artifact, space;
+
+    beforeAll(() => {
+        return jasmine.shelf.createSpace().then((testSpace) => {
+            space = testSpace;
+        });
+    });
+    beforeEach(() => {
+        return space.uploadArtifact("LOL").then((a) => {
```
Comment (GoogilyBoogily) [inline spec/metadata.spec.js]: I know it probably doesn't matter, but I don't think you should use the variable name `a`. Maybe `testArtifact`? Similar to how you did it in the beforeAll.

File: spec/metadata.spec.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        return artifact.metadata.updateAll(newMetadata).then(() => {
+            return artifact.metadata.getAll();
+        }).then((allMetadata) => {
+            expect(allMetadata.myNewProperty.value).toEqual(newMetadata.myNewProperty.value);
+            [
+                "md5Hash",
+                "artifactPath",
+                "artifactName"
+            ].forEach((prop) => {
+                expect(allMetadata[prop]).not.toBe();
```
Comment (GoogilyBoogily) [inline spec/metadata.spec.js]: If you don't want to put `null` in the `toBe()` because ESLint complains, you can use `toBeNull()`.

Comment (GoogilyBoogily) [inline spec/metadata.spec.js]: If you don't want to put `null` in the `toBe()` because ESLint complains, you can use `toBeNull()`.

**Review by GoogilyBoogily — CHANGES_REQUESTED** [review]

File: spec/helpers/lib/space.js (hunk 3 above)
Comment (kyle-long) [reply spec/helpers/lib/space.js]: I had a discussion with Tyler about this. He said if it is long he puts it in the description. Should we talk to him again?

File: spec/metadata.spec.js (hunk 6 above)
Comment (kyle-long) [reply spec/metadata.spec.js]: It was supposed to check it was not undefined. However, `toBeDefined` also exists

File: spec/metadata.spec.js (hunk 5 above)
Comment (kyle-long) [reply spec/metadata.spec.js]: changed

File: spec/helpers/lib/space.js (hunk 4 above)
Comment (kyle-long) [reply spec/helpers/lib/space.js]: fixed

File: spec/helpers/lib/space.js (hunk 2 above)
Comment (kyle-long) [reply spec/helpers/lib/space.js]: changed

File: spec/helpers/lib/space.js (hunk 3 above)
Comment (GoogilyBoogily) [reply spec/helpers/lib/space.js]: There's got to be a better way of handling that. Yeah, I think we should.

**Review by GoogilyBoogily — APPROVED** [review]
Looks good.

**Pull request merged**